packages/teams-js/src/internal/utils.ts

```diff
@@ -54,18 +62,28 @@ export function validateOrigin(messageOrigin: URL): boolean {
     return false;
   }
   const messageOriginHost = messageOrigin.host;
-
-  if (validOrigins.some((pattern) => validateHostAgainstPattern(pattern, messageOriginHost))) {
-    return true;
+  let isValid = Communication.validatedDomains.get(messageOrigin.host);
+  if (isValid === undefined) {
+    validateOriginHandler(messageOrigin).then((value: boolean) => {
```

Comment:
>then [](http://example.com/codeflow?start=41&length=4)

Should there be a handler here to handle the situation when the `Promise` from `validateOriginHandler` is rejected? At least to log that fact?